src/main/java/com/cloudbees/dockerpublish/DockerBuilder.java

```diff
@@ -267,33 +337,53 @@ private Result executeCmd(String cmd) throws IOException, InterruptedException {
             TeeOutputStream stdout = new TeeOutputStream(listener.getLogger(), baos);
             PrintStream stderr = listener.getLogger();
 
-            boolean result = launcher.launch()
-                    .envs(build.getEnvironment(listener))
-                    .pwd(build.getWorkspace())
-                    .stdout(stdout)
-                    .stderr(stderr)
-                    .cmdAsSingleString(cmd)
-                    .start().join() == 0;
+            // get Docker registry credentials
+            KeyMaterial registryKey = getRegistry().newKeyMaterialFactory(build).materialize();
+            // Docker server credentials. If server is null (right after upgrading) do not use credentials
+            KeyMaterial serverKey = server == null ? null : server.newKeyMaterialFactory(build).materialize();
```

Comment:
To be consistent you could set the `server` to be non-null in `readResolve` as well, but it does not really matter.